File: leetcode/9.palindrome-number.js
/*
 * @lc app=leetcode id=9 lang=javascript
 *
 * [9] Palindrome Number
 */

// @lc code=start
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
	var str = x.toString();
	if (str[0] == "-") return false;
	for (var i = 0; i < str.length / 2; i++) {
		if (str[i] != str[str.length - 1 - i]) return false;
	}
};
// @lc code=end

File: leetcode/9.palindrome-number.rs
/*
 * @lc app=leetcode id=9 lang=rust
 *
 * [9] Palindrome Number
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let mut sum = 0;
        let mut cur = x;
        while cur > 0 {
            sum = sum * 10 + cur % 10;
            cur /= 10;
        }

        sum == x
    }
}
// @lc code=end
